### ISSUE TITLE
[net9-rc2] Trim analysis warning IL2026 using XamlTypeResolver with Compiled Bindings

### ISSUE THREAD

**Issue by tranb3r** [dotnet/maui#25199]
### Description

When using compiled bindings, I get the following trim analysis warning:
```
D:\MauiAppTestTrimAnalysis\MauiAppTestTrimAnalysis\obj\Release\net9.0-android35.0\Microsoft.Maui.Controls.SourceGen\Microsoft.Maui.Controls.SourceGen.CodeBehindGenerator\MainPage.xaml.sg.cs(30,3): 
Trim analysis warning IL2026: 
MauiAppTestTrimAnalysis.MainPage.InitializeComponent(): 
Using member 'Microsoft.Maui.Controls.Xaml.Internals.XamlTypeResolver.XamlTypeResolver(IXmlNamespaceResolver, Assembly)' which has 'RequiresUnreferencedCodeAttribute' can break functionality when trimming application code. 
Loading XAML at runtime might require types and members that cannot be statically analyzed. 
Make sure all of the required types and members are preserved.
```

Is there a fix or workaround?

### Steps to Reproduce

1. Open repro app (simple app with one binding and x:datatype is set in xaml)
2. Build app in release mode
3. Check warnings

### Link to public reproduction project repository

https://github.com/tranb3r/Issues/tree/main/MauiAppTestTrimAnalysis

### Version with bug

9.0.0-rc.1.24453.9

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Affected platforms

Android

### Affected platform versions

Android 15

### Did you find any workaround?

No

**Comment by ninachen03** [comment]
This issue has been verified using Visual Studio 17.12.0 Preview 2.1(9.0.0-rc.2.24503.2 & 9.0.0-rc.1.24453.9). Can repro this issue at android platform.
![Image](https://github.com/user-attachments/assets/60341797-587f-4e6d-9f6f-1cf550b8c586)

**Comment by tranb3r** [comment]
@StephaneDelcroix @PureWeen @simonrozsival
This is still happening with VS 17.12.0 GA and net9 GA.
Could you please double check? Thanks

**Comment by jamescrosswell** [comment]
I'm [seeing something similar](https://github.com/getsentry/sentry-dotnet/issues/3785) trying to AOT compile the MAUI sample from the Sentry repository.

**Comment by simonrozsival** [comment]
@tranb3r I can repro the bug in .NET 9 GA. This should be fixed by #25420 already, but maybe it wasn't backported to GA and it will be release in the servicing release, @PureWeen?
